src/concrete/ml/sklearn/base.py

```diff
@@ -1993,23 +1995,7 @@ def scatter1d(x, v, indices):
 
         return numpy.expand_dims(topk_labels, axis=0)
 
-    # KNN works only for MONO in the latest concrete Python version
-    # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3978
     def compile(self, *args, **kwargs) -> Circuit:
-        # If a configuration instance is given as a positional parameter, set the strategy to
-        # multi-parameter
-        if len(args) >= 2:
-            configuration = force_mono_parameter_in_configuration(args[1])
-            args_list = list(args)
-            args_list[1] = configuration
-            args = tuple(args_list)
-
-        # Else, retrieve the configuration in kwargs if it exists, or create a new one, and set the
-        # strategy to multi-parameter
-        else:
-            configuration = kwargs.get("configuration", None)
-            kwargs["configuration"] = force_mono_parameter_in_configuration(configuration)
```

Comment:
I believe you can now remove `force_mono_parameter_in_configuration` from our source code !